### PR TITLE
Exclude shared pool tests

### PR DESCRIPTION
**Pull request by swatisehgal**
- Change cpuCount to cpuRequest in order to cater to cases  where guaranteed pods make non-integral CPU Requests.
- Add e2e tests for test cases for pods with non-integral CPUs

Signed-off-by: Swati Sehgal <swsehgal@redhat.com>

### CODE REVIEW

**Comment by AlexeyPerevalov** [comment]
in general ok, we had 2 options:
1. use non-integral
2. use different quantity for request/limits

I think, the first requires lesser code modification, so it preferred in test to archive a goal.
But, I wander why we can't keep float32 for cpuCount, in this case no need to modify every existing cpuCount field. MustParse works well with e.g. "1.5" and unassigned cpuCount would be parsed by MustParse as before. In match function it would be easier to compare.

**Comment by swatisehgal** [comment]
Good point Alexey, please take a look at the changes based on your comments. I don't think calling it `CPUCount` is appropriate as it  intrinsically means integral number of CPUs and that is why call it `CPURequest` but I have changed it to a float32 type. Let me know what you think.

File: test/e2e_node/podresources_test.go [hunk 1]
```diff
@@ -739,3 +777,11 @@ func enablePodResourcesFeatureGateInKubelet(f *framework.Framework) (oldCfg *kub
 
 	return oldCfg
 }
+
+func getCpuRequest(cpuQuantity float32) float32 {
```
Comment (AlexeyPerevalov) [inline test/e2e_node/podresources_test.go]: this function is no more needed

Comment (swatisehgal) [reply test/e2e_node/podresources_test.go]: Ah right!! My bad, fixing

**Comment by AlexeyPerevalov** [comment]
> Good point Alexey, please take a look at the changes based on your comments. I don't think calling it `CPUCount` is appropriate as it intrinsically means integral number of CPUs and that is why call it `CPURequest` but I have changed it to a float32 type. Let me know what you think.

I agree with naming

File: test/e2e_node/podresources_test.go [hunk 2]
```diff
@@ -182,12 +183,17 @@ func matchPodDescWithResources(expected []podDesc, found podResMap) error {
 			return fmt.Errorf("no container resources for pod %q container %q", podReq.podName, podReq.cntName)
 		}
 
-		if podReq.cpuCount > 0 {
-			if len(cntInfo.CpuIds) != podReq.cpuCount {
-				return fmt.Errorf("pod %q container %q expected %d cpus got %v", podReq.podName, podReq.cntName, podReq.cpuCount, cntInfo.CpuIds)
+		cpuRequest := getCpuRequest(podReq.cpuRequest)
+		if cpuRequest > 0 && cpuRequest == float32(int(cpuRequest)) {
```
Comment (AlexeyPerevalov) [inline test/e2e_node/podresources_test.go]: for the sake of readability
// we have something to check
if cpuRequest > 0 {
   // 1. integral case of guranted pod
   your code
   // 2. non-integral case of all other QoS types
if you dont't like comment, just introduce a local helper if isIntegral(cpuRequest) ;)
}

Comment (swatisehgal) [reply test/e2e_node/podresources_test.go]: Went with the helper  ;)

**Comment by swatisehgal** [comment]
@AlexeyPerevalov Just an FYI: https://github.com/kubernetes/kubernetes/pull/97415#issuecomment-921578151. Merge comits are not allowed so we need a bit of cleanup.

**Pull request merged**